smoke-test/tests/cypress/cypress/e2e/mutations/add_users.js

```diff
@@ -1,47 +1,100 @@
-const tryToSignUp = () => {
-    let number = Math.floor(Math.random() * 100000);
-    let name = `Example Name ${number}`;
-    cy.enterTextInTestId("email", `example${number}@example.com`);
-    cy.enterTextInTestId("name", name);
-    cy.enterTextInTestId("password", "Example password");
-    cy.enterTextInTestId("confirmPassword", "Example password");
-
-    cy.mouseover("#title").click();
-    cy.waitTextVisible("Other").click();
+const test_id = Math.floor(Math.random() * 100000);
+const email = `example${test_id}@example.com`;
+const password = "Example password Reset";
 
-    cy.get("[type=submit]").click();
-    return name;
+const tryToSignUp = () => {
+  let name = `Example Name ${test_id}`;
+  cy.enterTextInTestId("email", email);
+  cy.enterTextInTestId("name", name);
+  cy.enterTextInTestId("password", "Example password");
+  cy.enterTextInTestId("confirmPassword", "Example password");
+  cy.mouseover("#title").click();
+  cy.waitTextVisible("Other").click();
+  cy.get("[type=submit]").click();
+  return name;
 };
 
 describe("add_user", () => {
-    it("go to user link and invite a user", () => {
-        cy.login();
-
+  it("go to user link and invite a user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.waitTextVisible("Invite Users");
+    cy.clickOptionWithText("Invite Users");
+    cy.waitTextVisible(/signup\?invite_token=\w{32}/)
+      .then(($elem) => {
+        const inviteLink = $elem.text();
         cy.visit("/settings/identities/users");
-        cy.waitTextVisible("Invite Users");
+        cy.logout();
+        cy.visit(inviteLink);
+        let name = tryToSignUp();
+        cy.waitTextVisible("Welcome to DataHub");
+        cy.hideOnboardingTour();
+        cy.waitTextVisible(name);
+      })
+      .then(() => {
+        cy.logout();
+        cy.visit("/signup?invite_token=bad_token");
+        tryToSignUp();
+        cy.waitTextVisible("Failed to log in! An unexpected error occurred.");
+      });
+  });
+});
 
-        cy.clickOptionWithText("Invite Users");
+describe("Reset Password", () => {
+  it("Verify you can’t generate a reset password link for a non-native user", () => {
+    cy.login();
+    cy.visit("/settings/identities/users");
+    cy.wait(2000);
```

Comment:
Please avoid using fixed waits. This can cause test to become flaky. Wait for appearance of elements on the UI instead